### ISSUE TITLE
[BUG] References to definitions show up in OpenAPI doc 

### ISSUE THREAD

**Issue by robin-glimp** [JaredCE/serverless-openapi-documenter#85]
**Describe the bug**
I'm trying to get my APIs all documented with custom request / response bodies. Some of my responses are direct ports of AWS specific types. I can run `npx ts-json-schema-generator` to generate the correct definitions. Very useful! But the result isnt fully as I expected.

**To Reproduce**
Add this file as your model
```
temp.json

{
  "$ref": "#/definitions/AWSObject",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "AWSObject": {
      "additionalProperties": false,
      "properties": {
        "Name": {
          "type": "string"
        },
        "Status": {
          "type": "string"
        }
      },
      "type": "object"
    }
  }
}
```
Add the model to `custom.documentation.models` in the `serverless.ts` (yes I'm using json wrapped in TS)
```
serverless.ts

...
        {
          name: 'temp',
          contentType: 'application/json',
          schema: '${file(temp.json)}',
        }
...
```

Add model to the documentation block of a function
```
        {
          statusCode: 201,
          responseBody: {
            description: 'Temp response',
          },
          responseModels: {
            'application/json': 'temp',
          },
        },
```

Run the generate command: `serverless openapi generate --output openapi.json --format json`

The resulting openapi.json contains this schema:
```
{
  "components": {
    "schemas": {
      "temp": {
        "properties": {
          "AWSObject": {
            "additionalProperties": false,
            "properties": {
              "Name": {
                "type": "string"
              },
              "Status": {
                "type": "string"
              }
            },
            "type": "object"
          }
        }
      }
    }
  }
}

```

**Expected behavior**
I expect the result to be
```
{
  "components": {
    "schemas": {
      "temp": {
        "additionalProperties": false,
        "properties": {
          "Name": {
            "type": "string"
          },
          "Status": {
            "type": "string"
          }
        },
        "type": "object"
      }
    }
  }
}
```

**Desktop (please complete the following information):**
 - Serverless version: 3.27.0
 - serverless-openapi-documenter version: 0.0.43

**Additional context**
Yes I know I can just take all the properties from the definition, replace the reference and paste them there. But given that the JSONSchema with a top-level ref only seems to accept the properties of the object, not the name of the object itself, I think there might be a bug.

**Comment by robin-glimp** [comment]
Ah, looks like the `json-schema-ref-parser` library has several tickets with issues on top level refs..

See https://github.com/APIDevTools/json-schema-ref-parser/issues/201

**Comment by JaredCE** [comment]
Hey @robin-glimp, so yep, the `json-schema-ref-parser` library has issues with top level refs which is a little annoying, it's also going through a phase of transition of ownership too which may bring about some of the fixes.  

So with what we know about that library, I added some defensive coding around it here: https://github.com/JaredCE/serverless-openapi-documenter/blob/main/src/definitionGenerator.js#L477

So, what's going on is:

this schema
```
{
  "$ref": "#/definitions/AWSObject",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "AWSObject": {
      "additionalProperties": false,
      "properties": {
        "Name": {
          "type": "string"
        },
        "Status": {
          "type": "string"
        }
      },
      "type": "object"
    }
  }
}
```

is being changed by the library to have its `$ref` be `"$ref": "#"`.  I'm then using my defensive programming to check $ref is set to just `#` and then grabbing the contents of the old reference and putting them into the main properties.

Now I guess the issue is whether it should create a property of `AWSObject` or whether it should just put the properties from that into the main object without the property `AWSObject`?  I'm not 100% sure of the correct answer, I might need to defer to someone from the JSON schema community on that front.

**Comment by robin-glimp** [comment]
> Now I guess the issue is whether it should create a property of AWSObject or whether it should just put the properties from that into the main object without the property AWSObject? I'm not 100% sure of the correct answer, I might need to defer to someone from the JSON schema community on that front.

Neither do I. (Kinda new to this JSON Schema stuff).

My only reference to what is "correct", is this online tool on [https://www.jsonschemavalidator.net/](https://www.jsonschemavalidator.net/), which takes a schema and a JSON and validates it.

Here is the current outcome of the docs generator.
<img width="1652" alt="Scherm­afbeelding 2023-02-04 om 13 09 01" src="https://user-images.githubusercontent.com/92545486/216766856-f43ec26a-032d-43fc-ae28-2364f0ed6875.png">

Here is the expected outcome.
<img width="1654" alt="Scherm­afbeelding 2023-02-04 om 13 09 16" src="https://user-images.githubusercontent.com/92545486/216766870-d0861370-5e14-42fd-8695-0518f1bb1db1.png">

If you want to close this issue as this is not your code creating this potential bug, I would be ok with that. I'll leave it up to you.

**Comment by robin-glimp** [comment]
Actually.. Looking at your code a second time now.. 

[https://github.com/JaredCE/serverless-openapi-documenter/blob/main/src/definitionGenerator.js#L483](https://github.com/JaredCE/serverless-openapi-documenter/blob/main/src/definitionGenerator.js#L483) This line seems to add the `pathTitle` as the key of the properties object, being the core of my problems. Replacing that line with `Object.assign(deReferencedSchema, property)` seems to do the trick for my use case. 

The question indeed remains: what is the correct way of dereferencing according to JSON Schema spec. Do you have any contacts for that?

**Comment by JaredCE** [comment]
Yeah that's the issue.  I do have contacts (there's a JSON schema slack
group) and I'll try and ask them tomorrow as I'm out and about today.
Should have a fix/decision early this week. I suspect what I'm currently
doing is wrong, thinking about it, but will defer and see if I can get a
better answer.

**Comment by robin-glimp** [comment]
Awesome work! Can confirm this fixes my original problem with generating the docs and seeing the AWS type as a key.
